portfolio/src/main/webapp/script.js

```diff
@@ -29,61 +29,79 @@ function getComments() {
 			comments.forEach((comment) => {
 				let date = new Date(comment.date);
 				commentListElement.appendChild(
-					createListElement(comment.author, 
+					createListElement(comment.author,
 						date.getMonth() + '/' + date.getDate() + '/' +
 						date.getFullYear(), comment.content, comment.emotion));
 			})
-	    });
+		});
 }
 
 /**
  * Fetches delete-data, deletes all commennts
  */
-function deleteComments() {
+function deleteComments(author, date, text, emotion) {
 	const pageEl = document.getElementById("page");
 	const page = pageEl.value;
-	fetch('/delete-data?page=' + page, {
-		method: 'POST'
+	const queryStr = 'page=' + page + '&' +
+		'author=' + author + '&' +
+		'date=' + date + '&' +
+		'text=' + text + '&' +
+		'emotion=' + emotion;
+	fetch('/delete-data?' + queryStr, {
+		method: 'POST',
 	});
 
-	document.getElementById('comments-container').innerHTML = '';
+	if (text == 'undefined') {
```

Comment:
This comparison will return false when text is undefined. Here you'll want: `if (!text)`.

When calling `deleteComments()` with no arguments, `author`, `date`, `text`, and `emotion` will be undefined. When converting them to a string, they have the special behavior of being converted to `"undefined"`, which is what you depend on the data servlet. But comparing an undefined object to the string `"undefined"` will not cause the undefined object to be converted to a string before comparison.

Reply:
You can also do: `if (text === undefined)` to compare the text to the global `undefined` variable: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined

Using '===' over '==' is in the style guide: https://google.github.io/styleguide/jsguide.html#features-equality-checks

Reply:
Got it! Thanks